src/lib.rs

```diff
@@ -105,21 +127,21 @@ macro_rules! define_vector {
     };
 }
 
-define_vector!(Vec2, 8, f32, 2 <- "Vector of 2 f32s. Alignment 8, size 16.");
-define_vector!(Vec3, 16, f32, 3 <- "Vector of 3 f32s. Alignment 16, size 24.");
-define_vector!(Vec4, 16, f32, 4 <- "Vector of 4 f32s. Alignment 16, size 32.");
-define_vector!(DVec2, 16, f64, 2 <- "Vector of 2 f64s. Alignment 16, size 32.");
-define_vector!(DVec3, 32, f64, 3 <- "Vector of 3 f64s. Alignment 32, size 48.");
-define_vector!(DVec4, 32, f64, 4 <- "Vector of 4 f64s. Alignment 32, size 64.");
-define_vector!(UVec2, 8, u32, 2 <- "Vector of 2 u32s. Alignment 8, size 16.");
-define_vector!(UVec3, 16, u32, 3 <- "Vector of 3 u32s. Alignment 16, size 24.");
-define_vector!(UVec4, 16, u32, 4 <- "Vector of 4 u32s. Alignment 16, size 32.");
-define_vector!(IVec2, 8, i32, 2 <- "Vector of 2 i32s. Alignment 8, size 16.");
-define_vector!(IVec3, 16, i32, 3 <- "Vector of 3 i32s. Alignment 16, size 24.");
-define_vector!(IVec4, 16, i32, 4 <- "Vector of 4 i32s. Alignment 16, size 32.");
+define_vector!(Vec2, [0, 1], mint::Vector2<f32>, [x, y], 8, f32, 2 <- "Vector of 2 f32s. Alignment 8, size 16.");
```

Comment:
This should be mint::Vector2, and let the macro fill the generic in.

Reply:
I wasn't able to get Rust to be happy when I tried this. Here's a playground link with roughly the issue I was having: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d903999ab9c64918d0d94e0bdfc160e6